autoload/asyncomplete/sources/flow.vim

```diff
@@ -66,6 +71,26 @@ function! asyncomplete#sources#flow#get_source_options(opts)
        \ })
 endfunction
 
+let s:flowbin_path_by_dir = {} " dir: <path to flow>
+
+function! s:resolve_flowbin_path(file, fallback)
+    let l:dir = fnamemodify(a:file, ':h')
+    if len(s:flowbin_path_by_dir) > 100
+      let s:flowbin_path_by_dir = {}
```

Comment:
please add a comment about why you are clearing the dir so that after 3 months when we look at this code again we know exactly why.